File: platform/android/dnsproxy/lib/src/main/java/com/adguard/dnslibs/proxy/ProxySettingsOverrides.java
package com.adguard.dnslibs.proxy;

import java.util.Objects;

/**
 * The subset of {@link DnsProxySettings} available for overriding on a specific listener
 */
public class ProxySettingsOverrides {
    private Boolean blockEch;

    /**
     * Set {@link DnsProxySettings#blockEch} overriding value.
     * If null, overriding is not applied.
     */
    public void setBlockEch(Boolean blockEch) {
        this.blockEch = blockEch;
    }

    /**
     * @return {@link DnsProxySettings#blockEch} overriding value
     */
    public Boolean getBlockEch() {
        return this.blockEch;
    }

    @Override
    public boolean equals(Object x) {
        if (this == x) return true;
        if (x == null || getClass() != x.getClass()) return false;
        ProxySettingsOverrides that = (ProxySettingsOverrides) x;
        return Objects.equals(this.blockEch, that.blockEch);
    }

    @Override
    public int hashCode() {
        return Objects.hash(blockEch);
    }
}

File: platform/windows/cs/Adguard.Dns/Adguard.Dns/Api/DnsProxyServer/Configs/ProxySettingsOverrides.cs
namespace Adguard.Dns.Api.DnsProxyServer.Configs
{
    /// <summary>
    /// The subset of <see cref="DnsProxySettings"/>
    /// available for overriding on a specific listener.
    /// (A managed mirror for <see cref="AGDnsApi.ag_proxy_settings_overrides"/>)
    /// </summary>
    public class ProxySettingsOverrides
    {
        /// <summary>
        /// Overrides <see cref="DnsProxySettings.BlockEch"/> if not null
        /// </summary>
        public bool? BlockEch { get; set; }
    }
}